src/Language/Fixpoint/Solver/PLE.hs

```diff
@@ -875,13 +1018,80 @@ evalApp γ ctx e0 es et
         else do
           return (Nothing, noExpand)
 
+evalApp _ ctx e0 es _
+  | EVar f <- dropECst e0
+  , Just rewrite <- M.lookup f (icSMTRewrites ctx)
+  = do
+    -- No need to check if the eta beta flag is on since icSMTRewrites is emoty
```

Comment:
It is not apparent to me why `icSMTRewrites` is empty at this point. Also, note the typo in the comment.

Also I don't see the relevance of emptiness for the case at hand.

Reply:
You can see that at line 275 that if the flag is not enabled we return an empty list.
The comment is justifying why I'm not checking if the etabeta flag is enabled

Reply:
Thanks. I'm deferring this conversation until I understand better the need for `icSMTRewrites`.